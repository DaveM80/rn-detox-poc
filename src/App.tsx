/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import ScreenNames from './enums/ScreenNames';
import Home from './screens/Home';

const Stack = createNativeStackNavigator();

function App() {
  return (
    <SafeAreaProvider>
      <NavigationContainer>
        <Stack.Navigator initialRouteName={ScreenNames.Home}>
          <Stack.Screen name={ScreenNames.Home} component={Home} />
        </Stack.Navigator>
      </NavigationContainer>
    </SafeAreaProvider>
  );
}

export default App;
