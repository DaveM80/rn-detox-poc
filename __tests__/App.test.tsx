import 'react-native';
import React from 'react';
import renderer, { type ReactTestRenderer } from 'react-test-renderer';
import App from '../src/App';

// Note: test renderer must be required after react-native.

it('renders correctly', () => {
  renderer.act(() => {
    renderer.create(<App />);
  });
});

it('has no visual changes', () => {
  let treeObj: ReactTestRenderer | undefined;
  renderer.act(() => {
    treeObj = renderer.create(<App />);
  });
  const treeObjJSON = (treeObj as ReactTestRenderer).toJSON();
  expect(treeObjJSON).toMatchSnapshot();
});
