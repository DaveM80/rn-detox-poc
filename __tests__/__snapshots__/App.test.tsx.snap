// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has no visual changes 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text>
    Hello World!!!
  </Text>
</RCTSafeAreaView>
`;
